biogeophys/FatesPlantHydraulicsMod.F90

```diff
@@ -480,8 +485,8 @@ subroutine InitPlantHydStates(site, cohort)
           ! Calculate the mean total potential (include height) of absorbing roots
 !          h_aroot_mean = h_aroot_mean + cohort_hydr%psi_aroot(j) + mpa_per_pa*denh2o*grav_earth*(-site_hydr%zi_rhiz(j))
           
-          cohort_hydr%th_aroot(j) = wrf_plant(aroot_p_media,ft)%p%th_from_psi(cohort_hydr%psi_aroot(j))
-          cohort_hydr%ftc_aroot(j) = wkf_plant(aroot_p_media,ft)%p%ftc_from_psi(cohort_hydr%psi_aroot(j))
+          cohort_hydr%th_aroot(j) = wrfa%p%th_from_psi(cohort_hydr%psi_aroot(j))
+          cohort_hydr%ftc_aroot(j) = wkfa%p%ftc_from_psi(cohort_hydr%psi_aroot(j))
```

Comment:
Cray compiler (`cce/9.0.2-classic`) errors out on this reference:
```

          cohort_hydr%th_aroot(j) = wrf_plant(aroot_p_media,ft)%p%th_from_psi(cohort_hydr%psi_aroot(j))
          ^
ftn-959 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 483, Column = 11
  Function "WRF_PLANT", defined at line 483 (/gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90) must have an explicit interface because the result has the DIMENSION attribute.
                                                                ^
ftn-408 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 483, Column = 65
  This POINTER component is to the right of a part-ref with nonzero rank.
                                                                ^
ftn-408 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 483, Column = 65
  This POINTER component is to the right of a part-ref with nonzero rank.

          cohort_hydr%ftc_aroot(j) = wkf_plant(aroot_p_media,ft)%p%ftc_from_psi(cohort_hydr%psi_aroot(j))
          ^
ftn-959 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 484, Column = 11
  Function "WKF_PLANT", defined at line 484 (/gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90) must have an explicit interface because the result has the DIMENSION attribute.
                                                                 ^
ftn-408 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 484, Column = 66
  This POINTER component is to the right of a part-ref with nonzero rank.
                                                                 ^
ftn-408 crayftn: ERROR INITPLANTHYDSTATES, File = /gpfs/mira-home/azamatm/repos/E3SM/components/clm/src/external_models/fates/biogeophys/FatesPlantHydraulicsMod.F90, Line = 484, Column = 66
  This POINTER component is to the right of a part-ref with nonzero rank.

```
Adding a temp lets the builds pass. Also, references are not dependent on loop indexes, so taking them outside of the loop should be faster.